### PR TITLE
Align network client tests with core repo

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

When the network client tests in this repo were copied over to the core repo, some alterations were made. This commit copies them back to this repo so that the two test suites are easier to compare to each other. Changes include:

* Extracting `testsFor*` functions (functions that bundle similar tests together) into separate files
* Reordering the tests defined in `testsForRpcMethodsThatCheckForBlockHashInResponse`
* Reordering the set of RPC methods listed in `shared-tests.js`
* Reordering the type definitions defined in `shared-tests.js` to be closer to the functions that use them
* Updating the tests defined in `testsForRpcMethodSupportingBlockParam` so that when they make requests they pass params, even if it's just an empty array

For more context, see: https://github.com/MetaMask/core/pull/1087

## Manual Testing Steps

This commit makes no functional changes — it only makes changes to tests. All of the tests should pass, but if you want to manually verify, run:

```
yarn jest app/scripts/controllers/network/{createInfuraClient.test.js,createJsonRpcClient.test.js}
```

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by DigitalsupportTeam** [comment]
We'd be happy to look into this for you
kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

File: app/scripts/controllers/network/provider-api-tests/helpers.js [hunk 1]
```diff
@@ -177,24 +143,38 @@ function mockRpcCall({ nockScope, request, response, error, delay, times }) {
   // eth-query always passes `params`, so even if we don't supply this property,
   // for consistency with makeRpcCall, assume that the `body` contains it
   const { method, params = [], ...rest } = request;
-  const httpStatus = response?.httpStatus ?? 200;
-  let completeResponse;
+  let httpStatus = 200;
+  let completeResponse = { id: 2, jsonrpc: '2.0' };
```
Comment (Gudahtt) [inline app/scripts/controllers/network/provider-api-tests/helpers.js:147]: Hmm, why default to an id of `2` here?

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: app/scripts/controllers/network/provider-api-tests/helpers.js (hunk 1 above)
Comment (mcmire) [reply app/scripts/controllers/network/provider-api-tests/helpers.js:147]: Oops, I just now noticed this. I suspect this was copied from the core tests, although, I am not sure. I will look into this more tomorrow.

Comment (mcmire) [reply app/scripts/controllers/network/provider-api-tests/helpers.js:147]: Okay, I was right, this appears to have come from the changes that Zach made. As to why exactly 2 was chosen, I am not sure. Perhaps to stand out.

**Review by BelfordZ — APPROVED** [review]

**Pull request merged**